include/circt/Dialect/FIRRTL/FIRRTLExpressions.td

```diff
@@ -242,6 +242,26 @@ def SubaccessOp : FIRRTLExprOp<"subaccess", [
   let hasCanonicalizeMethod = true;
 }
 
+def MultibitMuxOp : FIRRTLExprOp<"multibit_mux"> {
```

Comment:
Should we just generalize MuxPrimOp instead of making a new op?

Reply:
:+1: for generalizing `MuxPrimOp` to avoid adding redundant operations. The thing we need to be careful about is how this interacts with Scala FIRRTL passes like `InferWidths` and `InferResets` that were originally written against a 2-way mux. I don't think this is a problem, and most of the canonicalizations and the width inference business should easily extend to N-way muxes. But it's good to bear in mind.

Reply:
That makes sense. I'll add support for InferWidts and InferResets. I think one tricky point is FIREmitter which would require to lower n-bit mux into 2-way mux.

Reply:
Yeah that's a good point. It would be great if `FIREmitter` knew how to deal with multiway muxes, but in the spirit of not having this blocked on some rarely-used feature, it might also be okay to just error out in the emitter if it encounters a non-2-way mux.

Reply:
As long as we form multiway muxes after InferWidths and InferResets, we should be good for the time being.  You should file an issue to add that support so it doesn't get lost (same with FIREmitter).